lambda/create-case/uid.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+)
+
+var (
+	a   = 53
+	c   = 11
+	max = 9999_9999_999
+)
+
+var dammTable = [10][10]byte{
+	{0, 3, 1, 7, 5, 9, 8, 6, 4, 2},
+	{7, 0, 9, 2, 1, 5, 4, 8, 6, 3},
+	{4, 2, 0, 6, 8, 7, 1, 3, 5, 9},
+	{1, 7, 5, 0, 9, 8, 3, 4, 2, 6},
+	{6, 1, 2, 3, 0, 4, 5, 9, 7, 8},
+	{3, 6, 7, 4, 2, 0, 9, 5, 8, 1},
+	{5, 8, 6, 9, 7, 2, 0, 1, 3, 4},
+	{8, 9, 4, 5, 3, 6, 2, 0, 1, 7},
+	{9, 4, 3, 8, 6, 1, 7, 2, 0, 5},
+	{2, 5, 8, 1, 4, 3, 6, 7, 9, 0},
+}
+
+// formatUID converts a sequence number into a formatted UID.
+func formatUID(n int) string {
+	if n > max {
+		panic("maximum uid size exceeded")
+	}
+
+	mixed := (a*n + c) % int(max)
```

Comment:
I think `a` and `c` need to be much larger: `a` so that two LPAs created in a similar timeframe are suitably distinct and `c` so that we don't end up with all the early LPAs starting with lots of 0s.

But I'm also not sure this is sufficient to prevent predictably: the jump is consistently `a` so it's easy to enumerate them even if that seed was hidden.

Reply:
The ADR says "our approach should inject sufficient padding to make iteration attacks easily detectable and avoid predictable step jumps"

You previously suggested [an LCG](https://en.wikipedia.org/wiki/Linear_congruential_generator), which would basically be the same maths except `n` is the previous ID rather than the index. That would lead to much less predictable and more spaced out numbers, we just need to make sure `a`, `c` and `max` won't allow it to loop.

e.g. LCG with 53 as index starting at `M-8368-7667-8211`:
- M-3544-6394-5575
- M-7865-8911-5391
- M-6892-2311-6081

Reply:
Please correct me if I'm wrong, but I think that would not work for what we want here as it would not go through the entire set:

```go
package main

import "fmt"

const (
	a = 7
	c = 3
	n = 10
)

func main() {
	prev := 0
	for i := 0; i < 10; i++ {
		prev = lcg(prev)
		fmt.Println(i, lcg(i), prev)
	}
}

func lcg(seed int) int {
	return (a * seed + c) % n
}
```

```
❯ go run main.go
0 3 3
1 0 4
2 7 1
3 4 0
4 1 3
5 8 4
6 5 1
7 2 0
8 9 3
9 6 4
```